src/broadcast/filelistener.cpp

```diff
@@ -1,26 +1,52 @@
 
 #include <QTextCodec>
 #include <QtConcurrentRun>
-#include <functional>
 
 #include "broadcast/filelistener.h"
+#include "broadcast/metadatafileworker.h"
 #include "preferences/metadatafilesettings.h"
 
 
 FileListener::FileListener(UserSettingsPointer pConfig)
-        : m_pFile(new QFile,[](QFile *file)->void{
-            file->resize(0);
-            delete file;
-          }),
-          m_COsettingsChanged(kSettingsChanged),
+        : m_COsettingsChanged(kSettingsChanged),
           m_pConfig(pConfig),
           m_latestSettings(MetadataFileSettings::getPersistedSettings(pConfig)) {
 
+    MetadataFileWorker *newWorker = new MetadataFileWorker(m_latestSettings.filePath);
```

Comment:
We want to get rid of all  plain "new" calls one day. Is there a reason that this is a pointer?

Reply:
What do you mean? Get rid of dynamic memory? And I copied this section from the qt docs. I assume that it needs to be allocated in the heap because otherwise it would be destroyed before the thread accesses it.

Reply:
No, there should be no heap memory required, because the owning thread lives longer than the thread. If it turns out that there is heap memory required, us a unique_ptr

Reply:
Yeah but if I allocate the object in the stack it gets destroyed when it goes out of scope.

Reply:
In the stack means as local variable.
You may consider to make it a member variable, in this case it becomes part of the parent object memory, Wich can be heap or stack.